keps/sig-cli/2953-kustomize-plugin-graduation/README.md

```diff
@@ -0,0 +1,764 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes converging Kustomize's various alpha extension mechanisms into a single KRM-driven feature that has an enhanced story around plugin distribution, discovery and trust. A central goal of this KEP is to equip Kustomize with an extension story that will be acceptable for full inclusion in `kubectl kustomize`. This KEP provides an overarching vision that ties together several existing proposals that go into more detail on specific features this KEP considers important to plugin graduation.
+
+
+
+### Key terminology
+
+*KRM*: [Kubernetes Resource Model](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md)
+
+*Plugin*: User-authored generator, transformer or validator (refers to both the "provider" program that implements it and the "config" YAML required to use it).
+
+*Plugin provider*: The program that implements the plugin (e.g. container, script, Go program). Analogous to the controller for a custom resource.
+
+*Plugin config*: The KRM-style YAML document that declares the desired state the plugin implements. In some plugin styles, it must also specify the plugin provider to execute and the specification to follow in doing so. Analogous to a custom resource object.
+
+## Motivation
+
+Kustomize currently supports two distinct categories of plugins, both of which are in alpha. These inconsistent enduring alphas cause [confusion and frustration](https://github.com/kubernetes-sigs/kustomize/issues/2721) for users and are a source of complexity for maintainers. See also [Plugin Compatibility Grid](#current-plugin-compatibility-grid).
+
+### Legacy plugins
+
+These are Kustomize's original plugin style, which was first proposed in the [Generators and Transformers KEP].  
+
+**kustomize/kubectl support**: These plugins work in recent versions of `kubectl kustomize` if the `--enable-alpha-plugins` is used.
```

Comment:
I am not sure kustomize flags make it all the up to kubectl
```
mbp ~ » kubectl --enable-alpha-plugins
Error: unknown flag: --enable-alpha-plugins
See 'kubectl --help' for usage
```

Reply:
Some of them do, in recent releases. I made a chart at the end of this document that shows what flags work where, in the current releases:

https://github.com/kubernetes/enhancements/blob/bef9b6c582c7f7451f4ff06ddfc90555a5618725/keps/sig-cli/2953-kustomize-plugin-graduation/README.md#current-plugin-compatibility-grid

Reply:
thank you so much!  Is the helm flag enabled in 1.22+?

Reply:
The helm flags are there in 1.21, but those aren't related to extensions since the helm implementation those flags govern is built-in.